### PR TITLE
[Compute] `az image builder create`: Add new parameters `--proxy-vm-size` and `--build-vm-identities` to support ProxyVmSize and UserAssignedIdentities customization 

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Description**

new parameters https://github.com/Azure/azure-cli/issues/20680
close https://github.com/Azure/azure-cli/issues/20680

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

**Review by wangzelin007 — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/_image_builder.py [hunk 1]
```diff
@@ -485,7 +485,9 @@ def create_image_template(  # pylint: disable=too-many-locals, too-many-branches
                                  namespace='Microsoft.Network', type='virtualNetworks', name=vnet,
                                  child_type_1='subnets', child_name_1=subnet)
         vnet_config = VirtualNetworkConfig(subnet_id=subnet)
-    vm_profile = ImageTemplateVmProfile(vm_size=vm_size, os_disk_size_gb=os_disk_size, vnet_config=vnet_config)
+    if subnet is not None and proxy_vm_size is not None:
+        vnet_config = VirtualNetworkConfig(subnet_id=subnet, proxy_vm_size=proxy_vm_size)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_image_builder.py:489]: May I ask can `VirtualNetworkConfig` be supported only when `subnet` and `proxy_vm_size` pass in at the same time?

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_image_builder.py:489]: If so, if users only pass in `proxy_vm_size`, do we need to consider raising an error to tell the user that `subnet` is missing?

Comment (PARADISSEEKR) [reply src/azure-cli/azure/cli/command_modules/vm/_image_builder.py:489]: Yes,this is required by the service https://github.com/Azure/azure-cli/issues/20680#issue-1075955660

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_image_builder.py:489]: OK, we can consider adding a check. When only `proxy_vm_size` is passed in, we can raising an error to prompt users `subnet` must also be passed in

File: src/azure-cli/azure/cli/command_modules/vm/_image_builder.py [hunk 2]
```diff
@@ -485,7 +486,12 @@ def create_image_template(  # pylint: disable=too-many-locals, too-many-branches
                                  namespace='Microsoft.Network', type='virtualNetworks', name=vnet,
                                  child_type_1='subnets', child_name_1=subnet)
         vnet_config = VirtualNetworkConfig(subnet_id=subnet)
-    vm_profile = ImageTemplateVmProfile(vm_size=vm_size, os_disk_size_gb=os_disk_size, vnet_config=vnet_config)
+    if proxy_vm_size is not None:
+        if subnet is not None:
+            vnet_config = VirtualNetworkConfig(subnet_id=subnet, proxy_vm_size=proxy_vm_size)
+        else:
+            raise ArgumentUsageError('Usage error: --proxy-vm-size is only configurable when --subnet is specified.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_image_builder.py:493]:
```suggestion
            raise RequiredArgumentMissingError('Usage error: --proxy-vm-size is only configurable when --subnet is specified.')
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**